### ISSUE TITLE
Add information tooltips to the "Term details" panel

### ISSUE THREAD

**Issue by hmenager** [edamontology/edam-browser#46]
Add the following information texts to the "Terms details" panel (HTML tooltips that appear when hovering an element) to help new users understand it better:

- [ ] "Term": "Preferred name for the concept"
- [ ] "Definition": "Short definition of the concept"
- [ ] "Comment": "Misc. information that may help understand the scope of the concept"
- [ ] "Exact synonyms": "Alternative term(s) that represent exactly the same concept"
- [ ] "Narrow synonyms": "Alternative term(s) that represent a slightly narrower scope"
- [ ] "Broad synonyms": "Alternative term(s) that represent a slightly broader scope"
- [ ] "Related synonyms": "Alternative term(s) that represent a closely overlapping scope"
- [ ] "URI": "Permanent concept identifier"
- [ ] "Parents": "Link(s) to the parent concept(s) which represent broader concepts"
- [ ] "Community usage": "Usage of this concept in various databases and registries"
- [ ] "bio.tools": "Bioinformatics Tools and Services Discovery Portal"
- [ ] "BioSphere": "IFB (ELIXIR France) Cloud Services to analyze life science data"
- [ ] "bioweb": "Online catalog of bioinformatics programs and databanks available at the Institut Pasteur"
- [ ] "TeSS": "ELIXIR Training Portal"
- [ ] "Links": "Links to this concept in other ontology browsers"

**Comment by raashika03** [comment]
Hi @hmenager  I would like to work on this issue as part of Outreachy. Can this be assigned to me?

**Comment by raashika03** [comment]
![Screenshot from 2021-03-30 06-40-52](https://user-images.githubusercontent.com/45493793/112918777-558b2f80-9123-11eb-8461-6e5da141ffd7.png)
Is this what you're looking for @hmenager ?

**Comment by matuskalas** [comment]
Exactly @raashika03, looks good!

(😁+1 to the dark mode🌑)

**Comment by raashika03** [comment]
![Screenshot from 2021-04-01 02-23-17](https://user-images.githubusercontent.com/45493793/113210240-acaf1280-9291-11eb-82df-6a1caee063f6.png)
is this better because in this Community Usage doesn't appears like a link but still it does have tooltip.
Removed that link cause it doesn't direct anywhere. But in this there's no indication that there's some extra information about Community usage unless the user stays at that word for sometime.
So with which, I should go. @matuskalas @hmenager ?

Sorry for updating little late on this issue, was getting some cache issue in the internet browser in this issue only, so the local changes wasn't reflecting. And it took me two days to figure out the same :(

**Comment by bryan-brancotte** [comment]
Hi @raashika03 

_I'm jumping in the issue_

"Community usage" is not clickable, but still it can have a tooltip, so definitly I would go for https://github.com/edamontology/edam-browser/issues/46#issuecomment-811464120

For 1.25, the version by itself is not a link, it is the popo that contains a link. So 1.25 should not be rendered as a link.

You indicate that the _user_ [have to] _stays at that word for sometime_ to see the tooltip, maybe there is other way of rendering a tooltip, way that allows you to reduce the amount of time before showing the tooltip ?

**Comment by raashika03** [comment]
> For 1.25, the version by itself is not a link, it is the popo that contains a link. So 1.25 should not be rendered as a link.
I meant that contains a pop up kind of thing but there's no indication of this on the main page, that just seems to a no. Unless you go on that itself.  So should anything be done to make that mark able that it contains a pop up. Will that be a good idea? I think no marking is also ok cause time of pop up is very fast. But just asking for another possibility 😁

> 
> You indicate that the _user_ [have to] _stays at that word for sometime_ to see the tooltip, maybe there is other way of rendering a tooltip, way that allows you to reduce the amount of time before showing the tooltip ?

Ha this sounds good to me. I'll explore other possibilities. Thank you @bryan-brancotte 😊

**Comment by bryan-brancotte** [comment]
I could be useful to have a way to indicate that a text is not a link, but still has tool-tip that can help. I don't know if there is something standard to address such issue, but finding one could be useful !

**Comment by raashika03** [comment]
> I could be useful to have a way to indicate that a text is not a link, but still has tool-tip that can help. I don't know if there is something standard to address such issue, but finding one could be useful !

Yes!  It seems a good idea, Thank you. I'll dig deeper into this.

**Comment by drashti171299** [comment]
this issue is assigned to someone @bryan-brancotte ?

**Comment by raashika03** [comment]
> this issue is assigned to someone @bryan-brancotte ?

Yes drishti I'm on this. I'll submit a pr in few days. I've completed around 70% task

**Comment by drashti171299** [comment]
okk no problem @raashika03

**Comment by matuskalas** [comment]
@raashika03 would you mind noting which bits are those that you haven't worked on yet, and let @drashti171299 do those?
(you could copy&paste the undone ones from the top of the issue into a comment down here)

**Comment by raashika03** [comment]
Yes of course @matuskalas, In the below image whatever isn't showing like a link is yet left to be done. Actually I've coded for rest also(which are in incomplete state) but not completely yet so that's in the middle state :smile: my exams are going on from 4th to 7th april, so until that time I won't be able to dedicate so much time to little tedious issues. And those which I left above in incomplete state are of that kind. So, I'll submit a successful PR for the same on 8th April. This's the only reason why my other PR's and issues are still in working state. I don't know if I should apologise for the delay :smile: But I'm trying to cover smaller issues which takes lesser time.    
![Screenshot from 2021-04-05 18-07-45](https://user-images.githubusercontent.com/45493793/113574568-fe60ef80-9639-11eb-8d13-ecf6b53a145e.png)

**Comment by matuskalas** [comment]
Thank you so much @raashika03! 🙏🏽 Please take some rest from EDAM until the 8th, definitely no need to apologise! 😸 Good luck with your exams! 🤞🏽

@drashti171299 would you like to fix the remaining ones, or a subset of them? I'm assigning you to this issue too 😉
The ones left are:
- [ ] "Definition": "Short definition of the concept"
- [ ] "Comment": "Misc. information that may help understand the scope of the concept"
- [ ] "Exact synonyms": "Alternative term(s) that represent exactly the same concept"
- [ ] "Narrow synonyms": "Alternative term(s) that represent a slightly narrower scope"
> - [ ] "Broad synonyms": "Alternative term(s) that represent a slightly broader scope"
> - [ ] "Related synonyms": "Alternative term(s) that represent a closely overlapping scope"
> These 2 only after added to the EDAM Broswer.
- [ ] "Parents": "Link(s) to the parent concept(s) which represent broader concepts"

**Comment by drashti171299** [comment]
yes @matuskalas  I would give my best to work on this issue.

**Comment by drashti171299** [comment]
@matuskalas  do we need to provide any link to this 4 remaining tags?

**Comment by raashika03** [comment]
> @matuskalas do we need to provide any link to these 4 remaining tags?

Yes @drashti171299 but not only to 4,  7 are left @matuskalas mentioned 7above

**Comment by drashti171299** [comment]
I have added tooltips to all 7.

**Comment by drashti171299** [comment]
just asking about links that which links should I provide.

**Comment by raashika03** [comment]
Can you show your result once? No link is needed to be added anywhere only tooltips should be added

**Comment by drashti171299** [comment]
okay then it is done.
![ec09669f-dac0-44ce-9977-182231332071](https://user-images.githubusercontent.com/52856536/113878889-a2cc6880-97d7-11eb-9d57-b1448eadf475.jpeg)

**Comment by raashika03** [comment]
Yes it's done you can create a pr for the same

**Comment by bryan-brancotte** [comment]
Hi @drashti171299 and @raashika03 

_I'm replying here for both #92 and #87 as the remarks are the same._

The goal here is to add tooltips on various word in the detail panel. Providing not only the tooltip, but the whole html elements make it more difficult to maintain, for example one of you used [`<font>`](https://github.com/edamontology/edam-browser/pull/92/files#diff-c11ae8cf18cba2e900b4eee3972122950380827115f0cfb873ad70ba03280aa6R251) while the other one user [`<div>`](https://github.com/edamontology/edam-browser/pull/87/files#diff-c11ae8cf18cba2e900b4eee3972122950380827115f0cfb873ad70ba03280aa6R254), also not both elements use `data-toggle=\"tooltip\"`. If now we want to change the way the tooltip are rendered, we would have to find each occurrences instead of only changing the code of the function generating the row. I recommend both of you to close your current PR, and start fresh new PR.

Here are some direction on how to properly solve this issue:
 * find the js code where a row is added in the panel
 * add a new argument which is the text of the tooltip
 * change the code of the method to add the tooltip when provided
 * Don't wait to have completed the issue to push, the sooner we see the code the sooner we can indicate you what should be changed, and thus the less time is consumed.

Best wishes

**Comment by drashti171299** [comment]
ok @bryan-brancotte . I will do changes as per your suggestion and update the code.

**Comment by drashti171299** [comment]
@bryan-brancotte I have updated the code as per your suggestions. let me know is there any change or not.

**Comment by raashika03** [comment]
@bryan-brancotte can be closed via #99
